src/server/comment-controller.js

```diff
@@ -0,0 +1,108 @@
+import express from 'express';
+import httpErrors from 'http-errors';
+import permissions from '../domain/permissions.js';
+import CommentService from '../services/comment-service.js';
+import DocumentService from '../services/document-service.js';
+import needsPermission from '../domain/needs-permission-middleware.js';
+import { validateBody, validateQuery } from '../domain/validation-middleware.js';
+import ClientDataMappingService from '../services/client-data-mapping-service.js';
+import { documentIdParamsOrQuerySchema } from '../domain/schemas/document-schemas.js';
+import { deleteCommentBodySchema, postCommentBodySchema, putCommentBodySchema } from '../domain/schemas/comment-schemas.js';
+
+const jsonParser = express.json();
+const { BadRequest, NotFound } = httpErrors;
+
+class CommentController {
+  static get inject() { return [CommentService, DocumentService, ClientDataMappingService]; }
+
+  constructor(commentService, documentService, clientDataMappingService) {
+    this.commentService = commentService;
+    this.documentService = documentService;
+    this.clientDataMappingService = clientDataMappingService;
+  }
+
+  async handleGetDocumentComments(req, res) {
+    const { documentId } = req.query;
+
+    const comments = await this.commentService.getAllDocumentComments(documentId);
+    const mappedComments = await this.clientDataMappingService.mapComments(comments);
+
+    return res.send({ comments: mappedComments });
+  }
+
+  async handlePutComment(req, res) {
+    const { user } = req;
+    const data = req.body;
+
+    const document = await this.documentService.getDocumentById(data.documentId);
+
+    if (!document) {
+      throw new BadRequest(`Unknown document id '${data.documentId}'`);
+    }
+
```

Comment:
Maybe we could check here also if the document is an internal one and throw if it isn't. Just to make sure...?